src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/NewPayloadV1Handler.cs

```diff
@@ -143,8 +143,19 @@ public async Task<ResultWrapper<PayloadStatusV1>> HandleAsync(ExecutionPayloadV1
                 return NewPayloadV1Result.Valid(block.Hash);
             }
 
-            bool parentProcessed = _blockTree.WasProcessed(parentHeader.Number, parentHeader.GetOrCalculateHash());
-            if (!parentProcessed)
+            BlockInfo parentBlockInfo = _blockTree.GetInfo(parentHeader.Number, parentHeader.GetOrCalculateHash()).Info;
+            bool parentProcessed = parentBlockInfo.WasProcessed;
+
+            // edge-case detected on GSF5 - during the transition we want to try process all transition blocks from CL client
+            // The last condition: !parentBlockInfo.IsBeaconBody will be true for terminal blocks. Checking _posSwitcher.IsTerminal might not be the best, because we're loading parentHeader with DoNotCalculateTotalDifficulty option
+            bool forceProcessing = !_poSSwitcher.TransitionFinished && (_blockTree.Head?.Number ?? 0) + 8 >= block.Number && !parentBlockInfo.IsBeaconBody;
```

Comment:
8?

Reply:
We're using 8 here too: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Facade/Eth/EthSyncingInfo.cs#L37

I should change this condition bool weAreCloseToHead = (_blockTree.Head?.Number ?? 0) + 8 >= block.Number;